pygfx/utils/show.py

```diff
@@ -16,9 +17,243 @@
+
+        Parameters
+        ----------
+        object : gfx.WorldObject
+            The object to show. If it is not a :class:`gfx.Scene <pygfx.Scene>`
+            then Display will wrap it into a new scene containing lights and a
+            background.
+        up : gfx.Vector3
+            If set, and ``object`` does not contain a controller, set the camera
+            controller's up vector to this value.
+
+        Notes
+        -----
+        If you want to display multiple objects prefer :class:`gfx.Group
+        <pygfx.Group>` over :class:`gfx.Scene <pygfx.Scene>` to avoid the pitfall of
+        forgetting to add lights to the scene.
+
+        """
+
+        if isinstance(object, Scene):
+            scene = object
+        else:
+            scene = Scene()
+            scene.add(object)
+
+            dark_gray = np.array((169, 167, 168, 255)) / 255
+            light_gray = np.array((100, 100, 100, 255)) / 255
+
+            background = Background(None, BackgroundMaterial(light_gray, dark_gray))
+            scene.add(background)
+            scene.add(AmbientLight(), DirectionalLight())
+        self.scene = scene
+
+        if len(self.find_children(Light)) == 0:
+            warnings.warn(
+                "Your scene does not contain any lights. Some objects may not be visible"
+            )
+
+        existing_cameras = self.find_children(Camera)
+        if self.camera:
+            pass
+        elif len(existing_cameras) > 0:
+            self.camera = existing_cameras[0]
+        elif not self.camera:
+            self.camera = PerspectiveCamera(70, 16 / 9)
+            self.camera.add(DirectionalLight())
+            # do we really want to modify the scene and add
+            # a camera?
+            self.scene.add(self.camera)
+
+        if self.renderer is None and self.canvas is None:
+            from wgpu.gui.auto import WgpuCanvas
+
+            self.canvas = WgpuCanvas()
+            self.renderer = WgpuRenderer(self.canvas)
+        elif self.renderer is not None:
+            self.canvas = self.renderer.target
+        elif self.canvas is not None:
+            self.renderer = WgpuRenderer(self.canvas)
+        elif self.canvas != self.renderer.target:
+            raise ValueError("Display's render target differs from it's canvas.")
+        else:
+            pass
+
+        controllers = self.find_children(Controller)
+        if self.controller is not None:
+            pass
+        elif len(controllers) > 0:
+            self.controller = controllers[0]
+        else:
+            look_at = self.camera.show_object(object)
+            self.controller = OrbitController(
+                self.camera.position.clone(), look_at, up=up
+            )
+            self.controller.add_default_event_handlers(self.renderer, self.camera)
+
+        self.canvas.request_draw(self.draw_function)
+        sys.modules[self.canvas.__module__].run()
+
+    # this should probably live on WorldObject
```

Comment:
Agreed

Reply:
@Korijn Should this happen in a separate PR where we move the function over, or do you want this to be part of this PR?

Reply:
Let's create a ticket and follow up later

Reply:
See https://github.com/pygfx/pygfx/issues/385